pkg/yurthub/proxy/remote/remote.go

```diff
@@ -125,3 +129,39 @@ func (rp *RemoteProxy) modifyResponse(resp *http.Response) error {
 	}
 	return nil
 }
+
+func (rp *RemoteProxy) errorHandler(rw http.ResponseWriter, req *http.Request, err error) {
+
+	klog.V(2).Infof("enter remote proxy error handler: %v. error: %v", util.ReqString(req), err)
+	if !rp.cacheMgr.CanCacheFor(req) {
+		klog.Errorf("can not cache for %s", util.ReqString(req))
+		rw.WriteHeader(http.StatusBadGateway)
+		return
+	}
+
+	ctx := req.Context()
+	if info, ok := apirequest.RequestInfoFrom(ctx); ok {
+		if info.Verb == "get" || info.Verb == "list" {
+			var errInfo error
+			obj, errQuery := rp.cacheMgr.QueryCache(req)
+			if errQuery == storage.ErrStorageNotFound {
+				reqInfo, _ := apirequest.RequestInfoFrom(req.Context())
+				errInfo = errors.NewNotFound(schema.GroupResource{Group: reqInfo.APIGroup, Resource: reqInfo.Resource}, reqInfo.Name)
+			} else if errQuery != nil {
+				klog.Errorf("failed to query cache for %s, %v", util.ReqString(req), errQuery)
+				errInfo = errors.NewInternalError(errQuery)
+			} else if obj == nil {
+				klog.Errorf("no cache object for %s", util.ReqString(req))
+				errInfo = errors.NewInternalError(fmt.Errorf("no cache object for %s", util.ReqString(req)))
+			}
+			if errInfo != nil {
+				util.Err(errInfo, rw, req)
+				return
+			}
+
+			util.WriteObject(http.StatusOK, obj, rw, req)
+			return
+		}
+	}
+	rw.WriteHeader(http.StatusBadGateway)
+}
```

Comment:
how about write some error message to rw?

Reply:
Refer to the default errHandler, do not write error messages. only return 502